io-engine/src/rebuild/rebuild_job_backend.rs

```diff
@@ -389,8 +431,9 @@ impl RebuildJobBackend {
             });
 
         let progress = (blocks_recovered * 100) / blocks_total;
+        assert!(progress < 100 || blocks_remaining == 0);
```

Comment:
```suggestion
        assert!(progress <= 100 || blocks_remaining == 0);
```
It may be called when it's complete.

Reply:
There was a bug in rebuild log management, making blocks_remaining > 0 even after all blocks were copied. So I added this assert to check that.

- progress < 100: blocks_remaining can be > 0, or =0 if  no "dirty" blocks remain.
- progress == 100: blocks_remaining must be 0, as all blocks were copied.

If it is like assert!(progress <= 100 || EXPR), EXPR is never checked as progress is always <= 100 due to the way it is calculated.

Reply:
Ah right sorry I had misread the statement previously, got it.